src/BuiltInTools/dotnet-watch/HotReload/AspNetCoreDeltaApplier.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ExternalAccess.DotNetWatch;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.DotNet.Watcher.Tools
+{
+    internal class AspNetCoreDeltaApplier : IDeltaApplier
+    {
+        private readonly IReporter _reporter;
+        private Task _task;
+        private NamedPipeServerStream _pipe;
+
+        public AspNetCoreDeltaApplier(IReporter reporter)
+        {
+            _reporter = reporter;
+        }
+
+        public async ValueTask InitializeAsync(DotNetWatchContext context, CancellationToken cancellationToken)
+        {
+            if (_pipe is not null)
+            {
+                _pipe.Close();
+                await _pipe.DisposeAsync();
+            }
+
+            _pipe = new NamedPipeServerStream("netcore-hot-reload", PipeDirection.InOut, 1, PipeTransmissionMode.Byte, PipeOptions.Asynchronous | PipeOptions.CurrentUserOnly);
+            _task = _pipe.WaitForConnectionAsync(cancellationToken);
+
+            if (context.Iteration == 0)
+            {
+                var deltaApplier = Path.Combine(AppContext.BaseDirectory, "hotreload", "Microsoft.Extensions.AspNetCoreDeltaApplier.dll");
+                context.ProcessSpec.EnvironmentVariables.DotNetStartupHooks.Add(deltaApplier);
+
+                // Configure the app for EnC
+                context.ProcessSpec.EnvironmentVariables["DOTNET_MODIFIABLE_ASSEMBLIES"] = "debug";
```

Comment:
What values can `DOTNET_MODIFIABLE_ASSEMBLIES` have?

Reply:
I think it's the only possible value for it: https://github.com/dotnet/runtime/issues/47274